src/main/java/io/cdap/plugin/servicenow/sink/service/ServiceNowSinkAPIRequestImpl.java

```diff
@@ -0,0 +1,272 @@
+              throw new IllegalStateException(String.format("Permission denied for '%s' operation.",
+                                                            config.getOperation()));
+            } else if (errorDetail.contains(ServiceNowConstants.INSERTION_ERROR)) {
+              LOG.warn("Error Response : {} ", decodedResponseBody);
+            } else if (errorDetail.contains((ServiceNowConstants.MAXIMUM_EXECUTION_TIME_EXCEEDED))) {
+              failedRequestId = servicedRequestsArray.get(i).getAsJsonObject().get(ServiceNowConstants.ID);
+            }
+          }
+        }
+
+        JsonArray unservicedRequestsArray = responseJSON.get(ServiceNowConstants.UNSERVICED_REQUESTS).getAsJsonArray();
+
+        if (unservicedRequestsArray.size() > 0) {
+          // Add failed request Id to unserviced requests
+          if (failedRequestId != null) {
+            unservicedRequestsArray.add(failedRequestId);
+          }
+
+          // Process unserviced requests array into unserviced requests map
+          Map<String, RestRequest> unservicedRequestsMap = processUnservicedRequestsArray(restRequestsMap,
+                                                                                          unservicedRequestsArray);
+          // Retry unserviced requests
+          createPostRequest(unservicedRequestsMap, accessToken);
+        }
+      }
+    } catch (IOException e) {
+      LOG.error("Error while connecting to ServiceNow", e.getMessage());
+      throw new RetryableException();
+    } catch (Exception e) {
+      throw e;
+    }
+  }
+
+  private String getErrorMessage(String responseBody) {
+    try {
+      JsonObject jo = gson.fromJson(responseBody, JsonObject.class);
+      return jo.getAsJsonObject(ServiceNowConstants.ERROR).get(ServiceNowConstants.MESSAGE).getAsString();
+    } catch (Exception e) {
+      return e.getMessage();
+    }
+  }
+
+  private ServiceNowBatchRequest getPayloadRequest(Map<String, RestRequest> restRequests) {
+    ServiceNowBatchRequest payloadRequest = new ServiceNowBatchRequest();
+    payloadRequest.setBatchRequestId(UUID.randomUUID().toString());
+    payloadRequest.setRestRequests(new ArrayList<>(restRequests.values()));
+    return payloadRequest;
+  }
+
+  /**
+   * Process the unserviced requests array into an unserviced requests map.    .
+   *
+   * @param restRequestsMap         The map of rest requests
+   * @param unservicedRequestsArray An array of unserviced requests
+   * @return unservicedRequestsMap The map of unserviced requests
+   */
+  private Map<String, RestRequest> processUnservicedRequestsArray(Map<String, RestRequest> restRequestsMap,
+                                                                  JsonArray unservicedRequestsArray) {
+    Map<String, RestRequest> unservicedRequestsMap = new HashMap<>();
+    for (int i = 0; i < unservicedRequestsArray.size(); i++) {
+      unservicedRequestsMap.put(unservicedRequestsArray.get(i).getAsString(),
```

Comment:
Store the request id in a variable, log an error an ignore request id if it's not found in restRequestsMap
something like this - 
```java
String requestId = unservicedRequestsArray.get(i).getAsString();
RestRequest request = restRequestsMap.get(requestId);
if (request == null) {
   // log error and continue
}
unservicedRequestsMap.put(requestId, request);
```

Reply:
done. 0a1c20a052aca8be095d4e1cde499b71fed53861